src/AWS.Deploy.Common/Utilities/DockerUtilities.cs

```diff
@@ -0,0 +1,86 @@
+    public static class DockerUtilities
+    {
+        /// <summary>
+        /// Gets the path of a Dockerfile if it exists at the default location: "{ProjectPath}/Dockerfile"
+        /// </summary>
+        /// <param name="recommendation">The selected recommendation settings used for deployment</param>
+        /// <param name="fileManager">File manager, used for validating that the Dockerfile exists</param>
+        /// <param name="dockerfilePath">Absolute path to the Dockerfile</param>
+        /// <returns>True if the Dockerfile exists at the default location, false otherwise</returns>
+        public static bool TryGetDefaultDockerfile(Recommendation recommendation, IFileManager? fileManager, out string dockerfilePath)
+        {
+            var dockerFileDirectory = new FileInfo(recommendation.ProjectPath).Directory?.FullName;
+            if (dockerFileDirectory == null)
+                throw new InvalidProjectPathException(DeployToolErrorCode.ProjectPathNotFound, "The project path is invalid when trying to detect a Dockerfile.");
+
+            dockerfilePath = Path.Combine(dockerFileDirectory, "Dockerfile");
+
+            if (fileManager == null)
+            {
+                fileManager = new FileManager();
+            }
+
+            if (fileManager.Exists(dockerfilePath))
+            {
+                return true;
+            }
+            else
+            {
+                dockerfilePath = string.Empty;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Gets the path of a the project's Dockerfile if it exists, from either a user-specified and the default location
+        /// </summary>
+        /// <param name="recommendation">>The selected recommendation settings used for deployment</param>
+        /// <param name="fileManager">File manager, used for validating that the Dockerfile exists</param>
+        /// <param name="dockerfilePath">Absolute path to a Dockerfile</param>
+        /// <returns>True if a Dockerfile is specified for this deployment, false otherwise</returns>
+        public static bool TryGetDockerfile(Recommendation recommendation, IFileManager? fileManager, out string dockerfilePath)
+        {
+            // Load the Dockerfile from the option setting. If one was found at the default location,
+            // assume it would be set here via the replacement token at this point.
+            var dockerFileOptionSetting = recommendation.GetOptionSetting("DockerfilePath");
+            dockerfilePath = recommendation.GetOptionSettingValue<string>(dockerFileOptionSetting);
+
+            if (!string.IsNullOrEmpty(dockerfilePath))
+            {
+                if (fileManager == null)
+                {
+                    fileManager = new FileManager();
+                }
+
+                // Double-check that it still exists in case it was move/deleted after being specified.
+                if (fileManager.Exists(dockerfilePath))
```

Comment:
Relative to project path.

Reply:
Throughout, switched to either an absolute or a relative path to the project directory.